java-lib/src/main/java/com/wavefront/ingester/DogStatsDDecoder.java

```diff
@@ -0,0 +1,81 @@
+package com.wavefront.ingester;
+
+import java.util.logging.Logger;
+import java.util.logging.Level;
+import java.util.Iterator;
+import java.util.List;
+import java.util.HashMap;
+import java.util.Map;
+
+import sunnylabs.report.ReportPoint;
+
+/**
+ * DogStatsD decoder that takes a string in this format:
+ *    metric.name:value|type|@sample_rate|#tag1:value,tag2
+ */
+public class DogStatsDDecoder implements Decoder {
+    private static final Logger LOG = Logger.getLogger(
+        DogStatsDDecoder.class.getCanonicalName());
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void decodeReportPoints(String msg, List<ReportPoint> out, String customerId) {
+        final Map<String, String> annotations = new HashMap<>();
```

Comment:
use a builder (ImmutableMap) for instance since you don't know the size at this point